express-notes-api/index.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const app = express();
+
+const fs = require('fs');
+
+app.use(express.json());
+
+// User can get an array of notes
+app.get('/api/notes', function (req, res) {
+  const arr = [];
+  fs.readFile('./data.json', function (err, data) {
+    if (err) { res.status(404).send('This note does not exist'); }
+    data = JSON.parse(data);
+    for (const keys in data.notes) {
+      arr.push(data.notes[keys]);
+    }
+    res.json(arr);
+  });
+});
+
+// User can GET a single note by id
+app.get('/api/notes/:id', function (req, res) {
+  const Id = parseInt(req.params.id);
+  fs.readFile('./data.json', 'utf8', function (err, data) {
+    if (err) { res.status(404).send('This note does not exist'); }
+    data = JSON.parse(data);
+    if (data.notes[Id] === undefined) {
+      res.status(404).send('This note does not exist');
+    } else if (data.notes[Id].id === Id) {
+      res.status(200);
+      res.json(data.notes[Id]);
+    }
+  });
+});
```

Comment:
These routes should respond with a `400 Bad Request` if the `id` the user provides isn't a positive integer. For example, a `GET /api/notes/234.1234asjkdfgh` request should result in a `400` response.

---

![image](https://user-images.githubusercontent.com/16143771/143333753-3716382d-0671-4b7e-8546-448f15d6d99b.png)